tests/integration/index.test.ts

```diff
@@ -61,42 +63,56 @@ describe('Process creation and deletion', () => {
   })
 
   describe('Sad path', () => {
-    let currentVersion: number
+    const validProcessName = '0'
+    let validVersionNumber: number
     before(async () => {
-      currentVersion = await getVersionHelper('0x30')
+      const currentVersion = await getVersionHelper('0x30')
+      validVersionNumber = currentVersion + 1
     })
 
     it('fails for invalid restriction name', async () => {
-      return assert.isRejected(createProcess('0', currentVersion + 1, invalidRestrictionName))
+      return assert.isRejected(
+        createProcess(validProcessName, validVersionNumber, noValidRestrictions),
+        NoValidRestrictionsError
+      )
     })
 
-    // it('fails for invalid restriction value', async () => {
-    //   return assert.isRejected(createProcess('0', currentVersion + 1, invalidRestrictionValue))
-    // })
+    it('fails for invalid restriction value', async () => {
+      return assert.isRejected(createProcess(validProcessName, validVersionNumber, invalidRestrictionValue), ZodError)
+    })
 
     it('fails for invalid json', async () => {
-      return assert.isRejected(createProcess('0', currentVersion + 1, 'invalidJson'))
+      return assert.isRejected(createProcess(validProcessName, validVersionNumber, 'invalidJson'), SyntaxError)
     })
 
     it('fails to create for same version', async () => {
-      return assert.isRejected(createProcess('0', currentVersion, validAllRestrictions))
+      return assert.isRejected(
+        createProcess(validProcessName, validVersionNumber - 1, validAllRestrictions),
+        VersionError
+      )
     })
 
     it('fails to create for too low version', async () => {
-      return assert.isRejected(createProcess('0', currentVersion - 1, validAllRestrictions))
+      return assert.isRejected(
```

Comment:
does is fail if function resolves?

Reply:
yes, if the function resolves it says `AssertionError: expected promise to be rejected but it was fulfilled with...`